modules/iam-types/src/main/scala/ch/epfl/bluebrain/nexus/commons/iam/types/identity/Identity.scala

```diff
@@ -0,0 +1,72 @@
+package ch.epfl.bluebrain.nexus.commons.iam.types.identity
+
+import akka.http.scaladsl.model.Uri
+import cats.Show
+import io.circe._
+import io.circe.generic.extras._
+import io.circe.generic.extras.semiauto._
+
+import scala.util.Try
+
+/**
+  * Base enumeration type for identity classes.
+  */
+sealed trait Identity extends Product with Serializable
+
+/**
+  * Represents identities that were authenticated from a third party origin.
+  */
+trait Authenticated {
+
+  /**
+    * @return the third party domain from where the authentication origins
+    */
+  def origin: Uri
```

Comment:
let's rename the field to `realm` to match the changes requested for IAM in BlueBrain/nexus-iam#36